streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -41,12 +41,12 @@ class Checkpoint(@transient ssc: StreamingContext, val checkpointTime: Time)
   val checkpointDuration = ssc.checkpointDuration
   val pendingTimes = ssc.scheduler.getPendingTimes().toArray
   val delaySeconds = MetadataCleaner.getDelaySeconds(ssc.conf)
-  val sparkConfPairs = ssc.conf.getAll
+  val sparkConfPairs = ssc.conf.getAll.filterNot { kv =>
```

Comment:
Maybe this can be turned into a generic function that removes given keys if the key is set in the config.